modules/core/src/copy.cpp

```diff
@@ -319,10 +319,10 @@ void Mat::copyTo( OutputArray _dst ) const
             Size sz = getContinuousSize(*this, dst);
             size_t len = sz.width*elemSize();
 
-#if defined HAVE_IPP
+#if defined HAVE_IPP && !(defined __LP64__ || defined _WIN64)
             CV_IPP_CHECK()
             {
-                if (ippiCopy_8u_C1R(sptr, (int)step, dptr, (int)dst.step, ippiSize((int)len, sz.height)) >= 0)
+                if (ippiCopy_8u_C1R(sptr, (int)step, dptr, (int)dst.step, ippiSize(len, sz.height)) >= 0)
```

Comment:
Issue's author does something wrong, because want to transfer 4+GB of memory. Definitely It is not the last function with similar problem.
Also this fix should be more usable: instead of fully disabled optimization we can add a check for 4Gb boundary:
```
if ((len * sz.height < 2UL << 32) && ippiCopy(...) >= 0)
```

BTW, For more stable behavior, I believe that Mat/UMat constructors should generate error for any 2GB+ allocations (may be under `#ifndef CV_FORCE_BIG_DATA`). @vpisarev Any thoughts?